### ISSUE TITLE
why make modifier = torch.arctanh(inputs * (2 - eps * 2) - 1 + eps)?

### ISSUE THREAD

**Issue by wanglinourin** [LinQinLiang/SSAH-adversarial-attack#3]
previous work usually said adv = data.cuda(), so I want to know why make modifier = torch.arctanh(inputs * (2 - eps * 2) - 1 + eps) and adv = 0.5 * (torch.tanh(modifier) + 1)? Thank you!

**Comment by lingjivoo** [comment]
This operation makes the optimization variable $ x+ delta $ follow that  0<= $ x+ delta $ <=1. We can think of this approach as a smoothing of clipped gradient descent that eliminates the problem of getting stuck in extreme regions. This operation was used in C&W  (Towards Evaluating the Robustness of Neural Networks) . You can found this at https://arxiv.org/pdf/1608.04644.pdf.
